lec17-rmarkdown.Rmd

```diff
@@ -1,4 +1,306 @@
+
+`r ''````{r, echo=FALSE, results='asis'}
+library(knitr)
+kable(head(mtcars))
+```
+
+You can also set the default data frame printing via the `df_print` option in
+your YAML metadata under `output` to do this automatically.
+
+```
+---
+title: Document
+output:
+    html_document:
+        df_print:kable
+---
+```
+
+#### In-line code
+
+If you want to state a value in your data in your text, it is best to reference
+to the actual variable or code containing that value rather than manually writing
+it out. Here is an example:
+
+There are `r ''``r nrow(df)` samples in this experiment.
+
+
+### Images and Figures
+To include an image, use the following syntax:
+```
+[caption for my image](path/to/image.png)
+```
+If you would like the caption you wrote to be included underneath your image,
+put the following in your YAML metadata:
+
+```
+---
+title: Document
+output:
+    html_document:
+        fig_caption: yes
+
+```
+As you've learned in doing your assignments, both the code and output of your
+`r` code chunk such as figures will show up in your output document. However,
+using code chunk options `echo` and `eval`, you can suppress output of the code
+underlying a graph, and only show the resulting plot from your code or vice versa.
+In this case, you probably want the former, and it would look like this:
+
+`r ''````{r, eval=TRUE, echo=FALSE}
+library(ggplot2)
+qplot(mpg, wt, data=mtcars)
+```
+
+You can also set the figure height and width too using the `fig.height` and
+`fig.width` options.
+
+`r ''````{r, fig.width=7, fig.height=7, eval=TRUE, echo=FALSE}
+library(ggplot2)
+qplot(mpg, wt, data=mtcars)
+```
+
+### In-Line Citations & Bibliography
+
+Pandoc can automatically generate citations and a bibliography in a number of
+styles. In order to use this feature, you will need to specify a bibliography
+file using the bibliography metadata field in a YAML metadata section.
+For example:
+
+```
+---
+title: "Sample Document"
+output: html_document
+bibliography: bibliography.bib
+---
+```
+
+Many bibliography formats are accepted (see the [R Markdown guides](http://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html)),
+such as a .bib file which many citation managers can generate for you and will
+hold all the references you need for your document (e.g. [EndNote](https://www.reed.edu/cis/help/LaTeX/EndNote.html))
+
+Every entry in your bibliography file should have a shorthand key id which
+when preceded by '@' allows you to reference the citation in-line. They usually
+go within square brackets and are separated by semicolons.
+
+```
+One Citation:
+Some fact [@Smith2014]
+
+Multiple Citations:
+Statement [@Smith2014; @Logan1997].
+```
+
+To make a bibliography, you may also want to specify a citation style guide to
+format your bibliography (in the form of a .csl file). The official repository
+for recognized citation styles is available here: https://github.com/citation-style-language/styles
```

Comment:
Mention that you can modify them and make your own style also due to the permissive licensing! This is a great tool for modifying styles http://editor.citationstyles.org/visualEditor/, I used it to modify one into my own liking once =p